File: ShoppingV2/app/src/main/java/dk/itu/shoppingv2/ItemsDB.java
package dk.itu.shoppingv2;

import java.util.ArrayList;
import java.util.List;

public class ItemsDB {
    private static ItemsDB sItemsDB;
    private List<Item> itemsDB= new ArrayList<>();;

    private ItemsDB() { fillItemsDB(); }

    public static ItemsDB get() {
        if (sItemsDB == null) sItemsDB= new ItemsDB();
        return sItemsDB;
    }

    public String listItems() {
        String r= "";
        for(Item i: itemsDB)
            r= r+"\n Buy " + i.toString();
        return r;
    }

    public void addItem(String what, String where){
        itemsDB.add(new Item(what, where));
    }

    public void fillItemsDB() {
        itemsDB.add(new Item("coffee", "Irma"));
        itemsDB.add(new Item("carrots", "Netto"));
        itemsDB.add(new Item("milk", "Netto"));
        itemsDB.add(new Item("bread", "bakery"));
        itemsDB.add(new Item("butter", "Irma"));
    }
}

File: ShoppingV4KotlinUnitTest/app/src/main/java/dk/itu/shoppingv4kotlinunittest/ItemsDB.kt
package dk.itu.shoppingv4kotlinunittest

import java.util.HashMap
import java.util.Observable

class ItemsDB private constructor() : Observable() {
    private val itemsDB = HashMap<String, String>()

    init {
        fillItemsDB()
    }

    companion object {
        private var sItemsDB: ItemsDB? = null

        fun get(): ItemsDB {
            return sItemsDB ?: ItemsDB().also{sItemsDB = it}
        }
    }

    fun listItems(): String {
        var r = ""
        for ((key, value) in itemsDB)
            r = "$r\n Buy $key in: $value"
        return r
    }

    fun size(): Int {
        return itemsDB.size
    }

    fun getWhere(what: String): String? {
        return itemsDB[what]
    }

    fun addItem(what: String, where: String) {
        itemsDB[what] = where
        setChanged()
        notifyObservers()
    }

    fun removeItem(what: String) {
        if (itemsDB[what] != null) itemsDB.remove(what)
        setChanged()
        notifyObservers()
    }

    private fun fillItemsDB() {
        itemsDB["coffee"] = "Irma"
        itemsDB["carrots"] = "Netto"
        itemsDB["milk"] = "Netto"
        itemsDB["bread"] = "bakery"
        itemsDB["butter"] = "Irma"
        setChanged()
        notifyObservers()
    }
}